style.md

```diff
@@ -151,6 +152,65 @@ An interface is two fields:
 If you want interface methods to modify the underlying data, you must use a
 pointer.
 
+### Verify Interface Compliance
+
+For types that are known or required to implement specific interfaces, add
+checks to verify interface compliance at compile time.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+type Handler struct{ /* ... */ }
+
+
+
+func (h *Handler) ServeHTTP(
+  w http.ResponseWriter,
+  r *http.Request,
+) {
+  ...
+}
+```
+
+</td><td>
+
+```go
+type Handler struct{ /* ... */ }
+
+var _ http.Handler = (*Handler)(nil)
+
+func (h *Handler) ServeHTTP(
+  w http.ResponseWriter,
+  r *http.Request,
+) {
+  // ...
+}
+```
+
+</td></tr>
+</tbody></table>
+
+The statement `var _ http.Handler = (*Handler)(nil)` will fail to compile if
+`*Handler` ever stops matching the `http.Handler` interface.
+
+Note that the interface was implemented on the pointer type `*Handler` above,
+which meant use of `(*Handler)(nil)` on the right side of `=`. If the
+interface was implemented on the value type, the right side would use the zero
+value of that type.
```

Comment:
since we're being pedantic - the zero value of a pointer type is `nil`, so the last sentence is true regardless of whether a pointer or value type is being asserted as implementing an interface - so it might be worth just standardizing this to something like:

> The right hand side of the assignment should be the zero value of the type being asserted - for pointers, this is `nil`; for a value type `T`, this is `T{}`: [optional side by side example]

I do think that a struct would be more illustrative than an integer type, though. Or both would be fine, but I'd at least include a struct since that's more syntactically confusing.